<launch>
  <rosparam param="ticks_meter">883.5778414</rosparam>
  <rosparam param="encoder_min">-2147483648</rosparam>
  <rosparam param="encoder_max">2147483647</rosparam>
  <rosparam param="base_frame_id">base_footprint</rosparam>
  <rosparam param="base_width">0.1730756</rosparam>

  <node pkg="differential_drive" type="pid_velocity.py" name="lpid_velocity" output="screen">
      <remap from="wheel" to="lwheel"/>
      <remap from="motor_cmd" to="lmotor_cmd"/>
      <remap from="wheel_vtarget" to="lwheel_vtarget"/>
      <remap from="wheel_vel" to="lwheel_vel"/>
      <rosparam param="Kp">1.0</rosparam>
      <rosparam param="Ki">0</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">-30</rosparam>
      <rosparam param="out_max">30</rosparam>
      <rosparam param="timeout_ticks">10</rosparam>
      <rosparam param="rolling_pts">5</rosparam>
      <rosparam param="rate">60</rosparam>

  </node>

  <node pkg="differential_drive" type="pid_velocity.py" name="rpid_velocity">
      <remap from="wheel" to="rwheel"/>
      <remap from="motor_cmd" to="rmotor_cmd"/>
      <remap from="wheel_vtarget" to="rwheel_vtarget"/>
      <remap from="wheel_vel" to="rwheel_vel"/>
      <rosparam param="Kp">1</rosparam>
      <rosparam param="Ki">0</rosparam>
      <rosparam param="Kd">-0.05</rosparam>
      <rosparam param="out_min">-30</rosparam>
      <rosparam param="out_max">30</rosparam>
      <rosparam param="timeout_ticks">10</rosparam>
      <rosparam param="rolling_pts">5</rosparam>
      <rosparam param="rate">60</rosparam>
  </node>


<node pkg="differential_drive" type="virtual_joystick.py" name="virtual_joystick" output="screen">  
<rosparam param="~rate">60</rosparam> 
<rosparam param="~timeout_ticks">5</rosparam>
</node>
  
<node pkg="differential_drive" type="twist_to_motors.py" name="twist_to_motors" output="screen">
  <rosparam param="~publish_rate">60</rosparam>
</node>
</launch>

